plugins/importexport/native/NativeImportExportPlugin.inc.php

```diff
@@ -132,12 +132,46 @@ function display($args, $request) {
 				if (in_array($document->documentElement->tagName, array('article', 'articles'))) {
 					$filter = 'native-xml=>article';
 				}
+
+				$deployment = new NativeImportExportDeployment($journal, $user);
+
 				libxml_use_internal_errors(true);
-				$content = $this->importSubmissions(file_get_contents($temporaryFilePath), $journal, $user, $filter);
+				$content = $this->importSubmissions(file_get_contents($temporaryFilePath), $filter, $deployment);
+				$templateMgr->assign('content', $content);
 				$validationErrors = array_filter(libxml_get_errors(), create_function('$a', 'return $a->level == LIBXML_ERR_ERROR ||  $a->level == LIBXML_ERR_FATAL;'));
 				$templateMgr->assign('validationErrors', $validationErrors);
 				libxml_clear_errors();
-				$templateMgr->assign('content', $content);
+
+				// Are there any submissions import errors
+				$processedSubmissionsIds = $deployment->getProcessedSubmissionsIds();
+				$submissionsErrors = array_filter($processedSubmissionsIds, create_function('$a', 'return !empty($a);'));
+				if (!empty($submissionsErrors)) {
+					$templateMgr->assign('submissionsErrors', $processedSubmissionsIds);
+				}
+				// Are there any issues import errors
+				$processedIssuesIds = $deployment->getProcessedIssuesIds();
+				$issuesErrors = array_filter($processedIssuesIds, create_function('$a', 'return !empty($a);'));
+				if (!empty($issuesErrors)) {
+					$templateMgr->assign('issuesErrors', $processedIssuesIds);
+				}
+				// If there are any submissions or validataion errors
+				// delete imported submissions and issues.
+				// Issues errors can be ignored here, because they only contain section mismatch errors,
+				// that shall only be displayed to the user but that do not influence the import objects.
+				if (!empty($submissionsErrors) || !empty($validationErrors)) {
```

Comment:
Just off the top of my head, do you think it makes sense to move the cleanup code into the deployment object?